python/mypy_protobuf.py

```diff
@@ -648,32 +735,52 @@ def _add_messages(messages, prefix, _fd):
             _add_enums(fd.enum_type, start_prefix, fd)
 
 
-def main():
-    # type: () -> None
-    # Read request message from stdin
-    if six.PY3:
-        data = sys.stdin.buffer.read()
-    else:
-        data = sys.stdin.read()
+def wrap_code_generator(func):
```

Comment:
look into contextlib.contextmanager - which makes this kind of thing ergonomic

```
@contextmanager
def wrap_code_generator():
    // do before code

    yield

    // do after code
```

Reply:
thanks. you are right. will change